src/Infrastructure/Setting/ProjectSettingsInitializer/ProjectSettingsInitializerProcessor.php

```diff
@@ -0,0 +1,189 @@
+use SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto;
+use SprykerSdk\Sdk\Core\Domain\Enum\ValueTypeEnum;
+use SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\ChangeDefaultValueQuestion;
+use SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\SettingValueQuestion;
+use SprykerSdk\Sdk\Infrastructure\Setting\SettingInitializerRegistry;
+use SprykerSdk\SdkContracts\Entity\SettingInterface;
+
+class ProjectSettingsInitializerProcessor implements ProjectSettingsInitializerProcessorInterface
+{
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\ChangeDefaultValueQuestion
+     */
+    protected ChangeDefaultValueQuestion $changeDefaultValueQuestion;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\SettingValueQuestion
+     */
+    protected SettingValueQuestion $settingValueQuestion;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Setting\SettingInitializerRegistry
+     */
+    protected SettingInitializerRegistry $settingInitializerRegistry;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\ProjectFilesInitializer
+     */
+    protected ProjectFilesInitializer $projectFilesInitializer;
+
+    /**
+     * @param \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\ChangeDefaultValueQuestion $changeDefaultValueQuestion
+     * @param \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\Question\SettingValueQuestion $settingValueQuestion
+     * @param \SprykerSdk\Sdk\Infrastructure\Setting\SettingInitializerRegistry $settingInitializerRegistry
+     * @param \SprykerSdk\Sdk\Infrastructure\Setting\ProjectSettingsInitializer\ProjectFilesInitializer $projectFilesInitializer
+     */
+    public function __construct(
+        ChangeDefaultValueQuestion $changeDefaultValueQuestion,
+        SettingValueQuestion $settingValueQuestion,
+        SettingInitializerRegistry $settingInitializerRegistry,
+        ProjectFilesInitializer $projectFilesInitializer
+    ) {
+        $this->changeDefaultValueQuestion = $changeDefaultValueQuestion;
+        $this->settingValueQuestion = $settingValueQuestion;
+        $this->settingInitializerRegistry = $settingInitializerRegistry;
+        $this->projectFilesInitializer = $projectFilesInitializer;
+    }
+
+    /**
+     * @param array<\SprykerSdk\SdkContracts\Entity\SettingInterface> $settings
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ProjectSettingsInitDto $projectSettingsDto
+     *
+     * @return array<\SprykerSdk\SdkContracts\Entity\SettingInterface>
+     */
+    public function initialize(array $settings, ProjectSettingsInitDto $projectSettingsDto): array
+    {
+        $settingsToSave = $this->handleSettingsValues($settings, $projectSettingsDto);
+        $this->initializeSettings($settings);
```

Comment:
it is not clear why we initialize `$settings` not `$settingsToSave`.

Reply:
If we change the default value the setting should be saved. But any setting should be initialized despite the default value.